src/core/Akka/Actor/ChildrenContainer/Internal/ChildrenContainerBase.cs

```diff
@@ -30,25 +54,29 @@ protected ChildrenContainerBase(IImmutableDictionary<string, IChildStats> childr
         public abstract IChildrenContainer ShallDie(IActorRef actor);
         public abstract IChildrenContainer Unreserve(string name);
 
-        public IReadOnlyList<IInternalActorRef> Children
+        public IReadOnlyCollection<IInternalActorRef> Children
         {
             get
             {
-                return (from stat in InternalChildren.Values
-                        let childRestartStats = stat as ChildRestartStats
-                        where childRestartStats != null
-                        select childRestartStats.Child).ToList();
+                var children = from stat in InternalChildren.Values
```

Comment:
it's kinda a personal thing but I find this linq syntax a little meh.
can we just do something like
`
return InternalChildren.Values
                    .OfType<ChildRestartStats>()
                    .Select(stats => stats.Child)
`